python/lsst/sims/catUtils/utils/ObservationMetaDataGenerator.py

```diff
@@ -131,8 +125,137 @@ def __init__(self, driver=None, host=None, port=None, database=None):
+        records from the OpSim database that match those constraints. If limit
+        is used, the first N records will be returned in the list.
+
+        Parameters
+        ----------
+        obsHistID :
+        expDate :
+        fieldRA :
+        fieldDec :
+        moonRa :
+        moonDec :
+        rotSkyPos :
+        telescopeFilter :
+        rawSeeing :
+        seeing :
+        sunAlt :
+        moonAlt :
+        dist2Moon :
+        moonPhase :
+        expMJD :
+        altitude :
+        azimuth :
+        visitExpTime :
+        airmass :
+        skyBrightness :
+        m5 :
+        boundType :
+        boundLength :
+        limit :
+
+        Returns
+        -------
+        `numpy.recarray` with OpSim records. The column names may be obtained as
+        res.dtype.names
+
+        .. notes:: The `limit` argument should only be used if a small example
+        is required.
+        """
+        query = self.baseQuery + ' FROM SUMMARY'
+
+        nConstraints = 0 # the number of constraints in this query
+
+        for column in self.columnMapping:
+            value = eval(column[0])
+            if value is not None:
+                if nConstraints > 0:
+                    query += ' AND'
+                else:
+                    query += ' WHERE '
+
+                if isinstance(value,tuple):
+                    if len(value)>2:
+                        raise RuntimeError('Cannot pass a tuple longer than 2 elements '+
+                                           'to getObservationMetaData: %s is len %d'
+                                           % (column[0], len(value)))
+
+                    # perform any necessary coordinate transformations
+                    if column[4] is not None:
+                        vmin = column[4](value[0])
+                        vmax = column[4](value[1])
+                    else:
+                        vmin = value[0]
+                        vmax = value[1]
+
+                    query += ' %s > %s AND %s < %s' % \
+                             (column[1], vmin, column[1], vmax)
+                else:
+                    # perform any necessary coordinate transformations
+                    if column[4] is not None:
+                        vv = column[4](value)
+                    else:
+                        vv = value
+                    query += ' %s == %s' % (column[1], vv)
+
+                nConstraints += 1
+
+        query += ' GROUP BY expMJD'
+
+        if limit is not None:
+            query += ' LIMIT %d' % limit
+
+        if nConstraints == 0 and limit is None:
+            raise RuntimeError('You did not specify any contraints on your query;' +
+                               ' you will just return ObservationMetaData for all poitnings')
+
+        results = self.opsimdb.execute_arbitrary(query, dtype=self.dtype)
+        return results
+
+    def getObservationMetaDataOld(self, obsHistID=None, expDate=None, fieldRA=None, fieldDec=None,
```

Comment:
I was just keeping it around for tests. So it sounds like you feel we can get rid of it so I will do that.